build_test/lib/src/builder.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build/build.dart';
+
+/// Copy the input asset to all possible output assets.
+void _defaultBehavior(
+        BuildStep buildStep, Map<String, List<String>> buildExtensions) =>
+    _copyToAll(buildStep, buildExtensions, (id) => id);
+
+void _copyToAll(BuildStep buildStep, Map<String, List<String>> buildExtensions,
+    AssetId Function(AssetId) readFrom) {
+  if (!buildExtensions.keys.any((e) => buildStep.inputId.path.endsWith(e))) {
+    throw new ArgumentError('Only expected inputs with extension in '
+        '${buildExtensions.keys.toList()} but got ${buildStep.inputId}');
+  }
+  for (final inputExtension in buildExtensions.keys) {
+    if (!buildStep.inputId.path.endsWith(inputExtension)) continue;
+    for (final outputExtension in buildExtensions[inputExtension]) {
+      final newPath = _replaceSuffix(
+          buildStep.inputId.path, inputExtension, outputExtension);
+      final id = new AssetId(buildStep.inputId.package, newPath);
+      buildStep.writeAsString(
+          id, buildStep.readAsString(readFrom(buildStep.inputId)));
+    }
+  }
+}
+
+FutureOr Function(BuildStep, Map<String, List<String>>) copyFrom(
```

Comment:
Lol wow I _really_ dislike this new syntax. Imo just use a typedef because it looks nicer and is `<1.x` compatible (even if we don't declare support for that I would prefer to be as compatible as possible).

Reply:
Done